packit_service/worker/testing_farm.py

```diff
@@ -67,84 +69,119 @@ def __init__(
         self.session.mount("https://", adapter)
         self.header: dict = {"Content-Type": "application/json"}
 
-    def run_testing_farm(self):
-        for chroot in self.tests_chroots:
-            pipeline_id = str(uuid.uuid4())
-            logger.debug(f"Pipeline id: {pipeline_id}")
-            payload: dict = {
-                "pipeline": {"id": pipeline_id},
-                "api": {"token": self.config.testing_farm_secret},
-            }
+    def report_missing_build_chroot(self, chroot: str):
+        self.status_reporter.report(
+            state="error",
+            description=f"No build defined for the target '{chroot}'.",
+            check_names=PRCheckName.get_testing_farm_check(chroot),
+        )
 
-            logger.debug(f"Payload: {payload}")
+    def run_testing_farm_on_all(self):
+        failed = {}
+        for chroot in self.tests_chroots:
+            result = self.run_testing_farm(chroot)
+            if not result["success"]:
+                failed[chroot] = result.get("details")
+
+        if not failed:
+            return HandlerResults(success=True, details={})
+
+        return HandlerResults(
+            success=False,
+            details={"msg": f"Failed testing farm targets: '{failed.keys()}'."}.update(
+                failed
+            ),
+        )
 
-            stg = "-stg" if self.config.deployment == Deployment.stg else ""
-            copr_repo_name = (
-                f"packit/{self.project.namespace}-{self.project.repo}-"
-                f"{self.event.pr_id}{stg}"
+    def run_testing_farm(self, chroot: str) -> HandlerResults:
+        if chroot not in self.tests_chroots:
+            # Who triggered that?
```

Comment:
naughty, naughty ;-)

Reply:
I don't want to add classical "this should not happen", but it should not really happen. I only want to be sure that we don't break anything in the future...